### PR TITLE
Support SyncOnReadVariable in mirrored strategy

### PR DESCRIPTION
**Pull request by vandanavk**
### Description of changes:
There is a new variable of type values.DistributedVariable called SyncOnReadVariable used for performance improvement in MirroredStrategy.
Users may face issues in case this new type of variable is used during distributed training.

Added tests for TF 2 keras fit with Mirrored strategy.

Also made some modifications to conftest and tests.sh. In TF, once we disable eager execution, we cannot re-enable eager execution. The modifications in these 2 files will ensure that all tests are run in eager mode first followed by non-eager mode.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/tensorflow/collection.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def add_for_mode(self, arg, mode=None):
             return self.add_variable(arg, mode=mode)
         elif isinstance(arg, tf.Tensor):
             return self.add_tensor(arg, mode=mode)
-        elif isinstance(arg, values.MirroredVariable):
+        elif isinstance(arg, (values.MirroredVariable, values.DistributedVariable)):
```
Comment (vandanavk) [inline smdebug/tensorflow/collection.py]: both MirroredVariable and SyncOnReadVariable are of type values.DistributedVariable. Probably could just check for values.DistributedVariable

Comment (NihalHarish) [reply smdebug/tensorflow/collection.py]: If both values.MirroredVariable and SyncOnReadVariable are of the type, then it probably makes sense to type check only the parent.

Also can you rename the `self.add_mirrored_variable`

Comment (vandanavk) [reply smdebug/tensorflow/collection.py]: done in the latest commit

**Comment by NihalHarish** [comment]
Add comments explaining the changes you have made in the test suite.

**Pull request merged**